hopr/hopr-lib/src/lib.rs

```diff
@@ -592,7 +592,14 @@ impl Hopr {
             me_onchain.public().to_address(),
         )));
 
-        info!("Creating chain components using provider URL: {:?}", cfg.chain.provider);
+        if cfg.chain.provider.is_none() {
+            info!("Creating chain components using default provider");
+        } else {
+            info!(
+                "Creating chain components using custom provider: {:?}",
+                cfg.chain.provider.clone().unwrap()
+            );
+        }
```

Comment:
```suggestion
        if let Some(provider) = cfg.chain.provider {
            info!("Creating chain components using custom provider: {provider}");
        } else {
            info!("Creating chain components using default provider");
        }
```

Reply:
Even better than this would be to actually move this into the `ChainNetworkConfig`, where also the default value is known.